### ISSUE TITLE
Default gateway is missing in 23.05

### ISSUE THREAD

**Issue by evoserver** [openwrt/openwrt#13598]
### Describe the bug

After upgrading 23.05 RC3 to RC4 on my TP-Link Archer C60, the default gateway is missing even if it's present in the network config file. The problem is present also in the final release

### OpenWrt version

r23482-7fe85ce1f2

### OpenWrt target/subtarget

ath79/generic

### Device

TP-Link Archer C60 v2

### Image kind

Official downloaded image

### Steps to reproduce

Simply upgrade to RC4 or to the final release

### Actual behaviour

No default gateway is present and WDS stopped working

### Expected behaviour

A default gateway should appear when issuing ip route

### Additional info

I solved by adding the following to /etc/rc.local
sleep 30
ip route add default via 192.168.2.1
192.168.2.1 is my gateway set in network configuration

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by evoserver** [comment]
This is the configuration file
`config interface 'loopback'
        option proto 'static'
        option ipaddr '127.0.0.1'
        option netmask '255.0.0.0'
        option device 'lo'

config globals 'globals'

config interface 'wan'
        option proto 'dhcp'
        option device 'eth1'

config interface 'wan6'
        option proto 'dhcpv6'
        option device 'eth1'

config interface 'lan'
        option proto 'static'
        option netmask '255.255.255.0'
        option ipaddr '192.168.50.1'
        option device 'br-lan'
        option defaultroute '0'

config switch
        option name 'switch0'
        option reset '1'
        option enable_vlan '1'

config switch_vlan
        option device 'switch0'
        option vlan '1'
        option ports '1 2 3 4 0t'

config interface 'wwan'
        option proto 'static'
        option netmask '255.255.255.0'
        list dns '192.168.2.5'
        option ipaddr '192.168.2.50'
        option gateway '192.168.2.1'

config interface 'repeater_bridge'
        option proto 'relay'
        list network 'lan'
        list network 'wwan'

config device
        option name 'br-lan'
        option type 'bridge'
        list ports 'eth0.1'`

and this is `ip ro sh`

192.168.2.0/24 dev phy1-sta0 scope link  src 192.168.2.50
192.168.50.0/24 dev br-lan scope link  src 192.168.50.1

**Comment by brada4** [comment]
try adding 
`option force_link '1'`
explicitly, it should be default and configure default route even when gateway does not show up immediately?

**Comment by evoserver** [comment]
It doesn't solve the problem. As you pointed out, force_link is on per default as shown in LuCI

**Comment by brada4** [comment]
Backup cond , reset it and build back piece by piece - i.e if tjere is smth amiss with migration

**Comment by lespinasse** [comment]
I'm afraid this bug is not getting the attention it deserves. I am seeing similar issues on a pi zero W, also configured as a wifi station. With -rc4 I can see that only the subnet route is being created (`10.1.0.0/16 dev phy0-sta0 scope link  src 10.1.0.50`). With -rc3 I can see that a default route is created too, as expected (`default via 10.1.0.1 dev phy0-sta0`). With the default route missing, -rc4 connectivity breaks with the rest of my network.

Here is my fairly minimal /etc/config/network:
```
config interface 'loopback'
        option device 'lo'
        option proto 'static'
        option ipaddr '127.0.0.1/8'

config interface 'lan'
        option proto 'static'
        option ipaddr '10.1.0.50/16'
        option gateway '10.1.0.1'
        list dns_search 'lespinasse.org'
        list dns '192.168.1.1'

config interface 'wired'
        option device 'eth0'                    # eth0 is external USB ethernet
        option proto 'static'
        option ipaddr '192.168.0.1/24'
```

The 'wired' interface is not in use most of the time, it's only there as a backup so I can plug an external USB ethernet to debug issues when wifi connectivity gets broken.

One common point between @evoserver's and my config is that we both use wifi as a station rather than an access point. We are on different hardware platforms so I suspect wifi STA mode might be widely broken in -rc4.

I could try reproducing/confirming the issue on a pi 4b if that helps (would it ?)

**Comment by evoserver** [comment]
I also noticed that WDS was missing so I reverted to RC3

**Comment by evoserver** [comment]
The problem is still present in the final release of 23.05.0

**Comment by danpawlik** [comment]
Could you explain better what is the issue?
After clean installation, i connect cable to wan, which gets gw from my ISP, then i connect via LAN and I get correct address from the dhcp server and everything is working normally.

So please be more verbose and write:what is your network topology.

**Comment by evoserver** [comment]
After clean installation, the system works, but after the configuration if I reboot, the default gateway is missing. I can connect to the repeater via WiFi, but I can't access the internet.
I repeat: I use this router as a repeater. It connects to the main router, which is the one connected to the internet. DHCP on the repeater isn't used as the address is fixed. However DHCP requested are routed to the main router via relayd interface.
I also noticed that WDS isn't working, just as it wasn't working with latest RCs of 22.03.0. WDS was fixed from 22.03.2 to 23.05.0-RC3.

**Comment by brada4** [comment]
"default gateway is missing" what do you get from `ip ro sh`

**Comment by brada4** [comment]
Fails on wired too.
If you are looking for workaround - DHCP client sets route, just set up permanent lease in dhcp server.

**Comment by danpawlik** [comment]
@evoserver  you never mentioned earlier that you use the router as a repeater so the title of the issue is not "default gateway is missing in 23.05" but "no default gateway after setup repeater". You did not describe bug properly, but on the reddit you are saying as nobody want to fix your bug.
I will try to do the tests during the weekend, but I don't believe that there is an issue in the 23.05 release, but somewhere in your configuration. On what docs/manuals/videos did you setup the repeater?

**Comment by evoserver** [comment]
@danpawlik I'm not on Reddit so I don't know what you are referring to. Gateway is set in the configuration file and that configuration always worked since 18.06. All the settings are written in my first reply. The only extra package installed is relayed. Now from 23.05-RC4 the gateway disappears at reboot and WDS stops working. If you need syslog and kernel log I can forward them to you.

**Comment by danpawlik** [comment]
Link for reddit: https://reddit.com/r/openwrt/s/nD7R9QhhTA

Now the situation is more clear. As I mentioned, I will try to reproduce that bug in few days. However, if the title will be more verbose, or explain better what is the main issue, for sure there would be someone, that have workaround or even patch for it

**Comment by lespinasse** [comment]
I was the one who mentioned it on reddit.

I think "default gateway is missing" describes well WHAT happens.

I am not clear as to WHEN it happens. @evoserver and I have two different configs that trigger the issue (both using wifi STA links). @brada4 mentionned he saw the issue on wired links too. Is the issue present in every setup that uses a statically defined gateway (as opposed to setting it up with DHCP) ?

**Comment by lespinasse** [comment]
Few experiments to try to figure out WHEN does this bug happen:

- Set up a wndr3800 system with similar /etc/config/network as [above](https://github.com/openwrt/openwrt/issues/13598#issuecomment-1749781795). once again this is with the lan interface being set up as a wifi client / STA. The default route does not get set up after a reboot.
- On the wndr3800 system, commented out the option gateway under interface 'lan', and added `option gateway '192.168.0.254'` under interface 'wired'. The default route does get set up as expected after a reboot.
- Still on the wndr3800 system, set up a 'br-wired' bridge device between ports eth0 and eth1, and modified the 'wired' interface to use the br-wired device rather than just eth0. The option gateway is still configured under the 'wired' interface. Once again, the default route does get set up as expected after a reboot.
- Removed the option gateway from the wired interface, and changed the lan interface to use option proto 'dhcp'. The default route does get set up as expected after a reboot.

So, as far as I can tell:
- I have observed the bug in all test configs using a combination of a wifi client / station interface, with option proto 'static' and a statically set option gateway. By 'observed the bug', I mean that the default route does not get set to the desired gateway after reboot. This covers @evoserver's initial report, as well as mine above on pi zero W, and my new experiments on wndr3800.
- The bug can be reproduced on a variety of hardware (most likely, any hardware that can be configured as a wifi client / station)
- The bug does not seem to happen (in my experiments) when using a wired interface, or a wireless client interface with option proto 'dhcp'.
- @brada4 reported observing the bug in a wired setup. I was not able to reproduce that in my experiments. @brada4, could you give us more details about the wired /etc/config/network which showed the issue ?

**Comment by fluttirci** [comment]
I also have this problem. But I use my ASUS RT-N18U router as a main router, connected to the GPON. No weird VPN configs, just bare default config I have. In 22.03.05 version everything was working. I upgraded to 23.05, it didn't connect to internet. I went back to 22.03.05, at there ip route brings a default line aswell. But in 23.05 no default line in ip route command. In 23.05 here my ip route goes like:
192.168.2.0/24 dev br-lan scope link src 192.168.2.1
And my /etc/config/network file is like this below. I have written personal data as SECRET. Also wan.35 is required for my ISP requires vlan id 35.

config interface 'loopback'
	option device 'lo'
	option proto 'static'
	option ipaddr '127.0.0.1'
	option netmask '255.0.0.0'

config globals 'globals'
	option packet_steering '1'
	option ula_prefix 'SOMEIPV6PREFIX'

config device
	option name 'br-lan'
	option type 'bridge'
	list ports 'lan1'
	list ports 'lan2'
	list ports 'lan3'
	list ports 'lan4'

config interface 'lan'
	option device 'br-lan'
	option proto 'static'
	option netmask '255.255.255.0'
	option ip6assign '60'
	option ipaddr '192.168.2.1'

config device
	option name 'wan'
	option macaddr 'SECRET'

config interface 'wan'
	option proto 'pppoe'
	option device 'wan.35'
	option username 'SECRET'
	option password 'SECRET'
	option ipv6 'auto'

config interface 'wan6'
	option device 'wan'
	option proto 'dhcpv6'

**Comment by brada4** [comment]
Dor me it does not set route if cable is disconnected at boot.

**Comment by paulcarroty** [comment]
Got this bug using [client howto](https://openwrt.org/docs/guide-user/network/wifi/connect_client_wifi). AP pings are OK, no internet.

- `option proto dhcp` - it works to the first disconnect or the lease end
- `option proto static` - no default route, AP pings are OK. The options `gateway` and `dns` aren't helpful.

**Comment by im-0** [comment]
I was able to circumvent this by creating `/etc/hotplug.d/iface/90-missing-default-route` with following script:

```bash
#!/bin/sh

if [ "${ACTION}" = "ifup" ] && [ "${INTERFACE}" = "wwan" ]; then
	ip route add default via 192.168.1.1 dev "${DEVICE}"
	logger -t fix "Default route added with return code $?"
fi
```

Just replace `wwan` with your interface name from `/etc/config/network` (`config interface 'wwan'`) and `192.168.1.1` with your gateway.

**Comment by paulcarroty** [comment]
Tested the default route config using [static routes guide -> second example](https://openwrt.org/docs/guide-user/network/routing/routes_configuration) and it failed - no route at all. Something in `/etc/config/network` parser is definitely broken.

**Comment by brada4** [comment]
> definetly broken

It has to be `option table 'main'` for certain as opposed to example (learned by trial and error, documentation is inconclusive at best)

**Comment by danpawlik** [comment]
@evoserver hey, that issue does not exists in 23.05 RC3, right? I'm checking the [changes](https://github.com/openwrt/openwrt/compare/v23.05.0-rc3..v23.05.0-rc4) between releases   and I don't see any clear answer why it happens. I suspect that something might be related to [netifid](https://github.com/openwrt/netifd). Hope I will get some time to dig into that

**Comment by lespinasse** [comment]
I can't speak about @evoserver's case, but mine appeared between 23.05 rc3 (worked) and rc4 (no default route). I agree that netifd looks like a prime suspect.

**Comment by evoserver** [comment]
@danpawlik I confirm that the issue doesn't exist in 23.05.0-RC3. In fact, I'm using that release right now.

**Comment by fluttirci** [comment]
Some more info on this, on the working 22.03.05, there is a device automatically created as "pppoe-wan" but on 23.05 no such device. See on the ss below. This is the 22.03.05, working one:
![çalışan devices kısmı](https://github.com/openwrt/openwrt/assets/83621008/b15e7bea-f1a0-4612-ba97-8189f1e5180f)
Here this is the 23.05, no pppoe-wan:
![çalışmayan devices kısmı](https://github.com/openwrt/openwrt/assets/83621008/07572719-b868-48af-941c-7a7b82fe8b3b)
Also this is the 22.03.05 again:
![çalışan interfaces](https://github.com/openwrt/openwrt/assets/83621008/0d63936e-faf1-4fdd-b931-ec994e46d310)
And the same page on 23.05:
![çalışmayan interfaces](https://github.com/openwrt/openwrt/assets/83621008/e5721b47-276f-4d43-9573-1673c27af9f1)

**Comment by brada4** [comment]
there is no automatic pppoe-wan unless you configure one.

**Comment by fluttirci** [comment]
> there is no automatic pppoe-wan unless you configure one.

I tried to create one in interface>devices page. But it won't recognize it properly. When I try to add ip route it won't work. But it will be created in etc/config/network. I think something else is wrong.

**Comment by marinelli** [comment]
I had the same problem with multiple routers where I configured one of the wifi interfaces in client mode (sta).
I fixed the problem when I noticed that the old device name (ie: `wlan0`) was not available anymore, even in 23.05.0-rc3:
```
 config interface 'wwan'
-	option device 'wlan0'
+	option device 'phy0-sta0'
 	option proto 'static'
```
After changing that, I upgrade from RC3 to RC4 as usual, and everything worked as expected.

If you take a look at the diff https://github.com/openwrt/openwrt/compare/v23.05.0-rc3...v23.05.0-rc4, you will find the following commit:
- [hostapd: use phy name for hostapd interfaces instead of first-bss ifname](https://github.com/openwrt/openwrt/commit/09b9d732ec76a0cf0c80223f1c0b991139dd4e82)

I think this is the root of all the problems reported here. Even if it just mention `hostapd`, the code above might change the behavior of some code shared with `wpa_supplicant` too (they are part of the _wpad_ package).

It would be nice to re-introduce, as fallback, the use of the _first-bss_ available, and/or report in some way that a wifi device specified in the configuration is not available anymore.

My two cents.

**Comment by lespinasse** [comment]
Ah, thanks @marinelli for the hint - I can confirm this works for me as a workaround.

My normal config as quoted [above](https://github.com/openwrt/openwrt/issues/13598#issuecomment-1749781795) does not include any option device within the config interface 'lan' /etc/config/network section. This hasn't been previously required, because /etc/config/wireless mentions the option network 'lan' in its config wifi-iface section. So the wireless code would come up with a device name, and add that newly created device within the 'lan' interface defined in /etc/config/ntework.

With 23.05 (starting with -rc4), this doesn't work implicitly anymore (the default route fails to get created), but adding an explicit option device 'phy0-sta0' in the /etc/config/network config interface 'lan' definition causes the default route to be created again.

I still think that the current behavior is wrong: the wifi interface name should not have to be specified in /etc/config/network if the wireless code is the one picking it. When using multiple wifi SSIDs, it's not that easy to guess what device names will be created for them. But regardless, your observation provides a workable workaround for at least the simpler configs, and maybe will help figure out where the handling code goes wrong...

**Comment by marinelli** [comment]
> I still think that the current behavior is wrong: the wifi interface name should not have to be specified in /etc/config/network if the wireless code is the one picking it. When using multiple wifi SSIDs, it's not that easy to guess what device names will be created for them. But regardless, your observation provides a workable workaround for at least the simpler configs, and maybe will help figure out where the handling code goes wrong...

If I remove the `device` option from the `wwan` interface configuration, than RC3 works, but RC4 is broken.
I agree that the current behavior makes a previous valid configuration, invalid: it does not allow anymore to specify a wifi network config without a device.
The re-introduction of the previous behavior in `wpad`, as a fallback method, would allow this kind of configurations (that might be helpful for creating default configurations).

In general I think that using/having more predictable names for network interfaces is always the best approach.

**Comment by lespinasse** [comment]
Regarding explicit / predictable names: I would love to be able to specify a wifi-iface name in /etc/config/wireless, and have /etc/config/network refer to it with option device, rather than having the wifi-iface pick an arbitrary name and refer to the /etc/config/network interfaces with option network. I think being able to specify my own wifi-iface names would help clarify things in the UI (for example, where luci shows which stations are attached to which wifi-iface, it woudld be nice to display the chosen name rather than one that was automatically generated). It would also help if one needs to refer to that wifi-iface name in firewall rules or the like. I think it would make a ton of sense, though that is likely a bigger change than one would want to make on a stable release branch.

**Comment by fgimenezm** [comment]
But configuring the interface name is already an option. I named mines wlan0, wlan1, wlan2, wlan-guest.Maybe that’s what’s broken?I’ll try later removing the custom names to see what happens El 22 oct 2023, a la(s) 10:50, Michel Lespinasse ***@***.***> escribió:﻿
Regarding explicit / predictable names: I would love to be able to specify a wifi-iface name in /etc/config/wireless, and have /etc/config/network refer to it with option device, rather than having the wifi-iface pick an arbitrary name and refer to the /etc/config/network interfaces with option network. I think being able to specify my own wifi-iface names would help clarify things in the UI (for example, where luci shows which stations are attached to which wifi-iface, it woudld be nice to display the chosen name rather than one that was automatically generated). It would also help if one needs to refer to that wifi-iface name in firewall rules or the like. I think it would make a ton of sense, though that is likely a bigger change than one would want to make on a stable release branch.

**Comment by paulcarroty** [comment]
Double bind of device to a network interface breaking DRY principle.

I'll test it later and will try to add the note to OWRT wiki.

**Comment by marinelli** [comment]
> But configuring the interface name is already an option. I named mines wlan0, wlan1, wlan2, wlan-guest.Maybe that’s what’s broken?I’ll try later removing the custom names to see what happens

If you rename the device, it works without problems.

Btw, I didn't know about it, thank you for sharing @fgimenezm.

**Comment by lespinasse** [comment]
I might be missing a step here - is there a wifi-iface option to set the name of the interface, other than letting the wifi code pick its own name like phy0-sta0 ?

**Comment by marinelli** [comment]
> I might be missing a step here - is there a wifi-iface option to set the name of the interface, other than letting the wifi code pick its own name like phy0-sta0 ?

`ifname`, see here https://openwrt.org/docs/guide-user/network/wifi/basic#common_options1

**Comment by fgimenezm** [comment]
ok, so I tested these in my wireless client setup with static IP. I rebooted every time after setting the options.

case 1: custom iframe in /etc/config/wireless AND option device in /etc/config/network referencing the ifname
Didn't work... still no default GW

case 2: removed the "option device" from  /etc/config/network and the custom ifname from /etc/config/wireless (basically letting openwork use the default names)
Didn't work... still no default GW

case 3: without custom ifname in /etc/config/wireless but with "option device" in /etc/config/network (default names and specific device in network)
Success! Default GW is there after a reboot.

There seems to be a bug with the "ifname" support then, and also the double bind of device and network interface as @paulcarroty mentioned

**Comment by jow-** [comment]
Directly referencing a wifi ifname from /etc/config/network never was a valid supported configuration and only worked by accident. The proper canonical was is using the network option in wifi-iface to "attach" a wireless network to an ip configuration:

```
# /etc/config/network 
config interface wwan
  option proto dhcp
  # note no device or ifname

# /etc/config/wireless
config wifi-iface
  ...
  option network wwan
```

**Comment by fgimenezm** [comment]
> Directly referencing a wifi ifname from /etc/config/network never was a valid supported configuration and only worked by accident. The proper canonical way is using the network option in wifi-iface to "attach" a wireless network to an ip configuration:
> 
> ```
> # /etc/config/network 
> config interface wwan
>   option proto dhcp
>   # note no device or ifname
> 
> # /etc/config/wireless
> config wifi-iface
>   ...
>   option network wwan
> ```

That's exactly what I did in the first place and the default gateway was not configured.
It started to work after I added "option device 'phy0-sta0'" to the network config.
And it only works with the default name: phy0-sta0. If I use a custom ifname the default gateway is not configured.

**Comment by nbd168** [comment]
Could somebody please test if this patch helps? https://nbd.name/p/d6151703

**Comment by lespinasse** [comment]
Hi Felix, I applied the patch to a built image (i.e. changed the .uc files in /usr/share/hostap), default route still missing after a reboot.

**Comment by fluttirci** [comment]
> This netifd update should fix it: https://git.openwrt.org/41d7439af5a40a71a0deb2075fa133b61b0e478a

How do I update this my dude? My internet doesn't work in my case. Also I am using ASUS RT-N18U, is this ok for my arch?

**Comment by jow-** [comment]
@fluttirci - your issue (missing `pppoe-wan`) is unrelated to the one handled in this ticket (wireless interfaces do not properly setup defualt route).

**Comment by rotanid** [comment]
this fix sounds like it would make this workaround obsolete:
https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=package/base-files/files/etc/init.d/boot;hb=HEAD#l47
thoughts about my theory?

... and therefore also finally fix our downstream issue at https://github.com/freifunk-gluon/gluon/issues/2779

**Comment by brada4** [comment]
How this issue is related to botched downstream upgrades?

**Comment by lespinasse** [comment]
Thank you Felix. I built pi-4 and pi-0-W images with the 2023-11-07-516ab774-1 netifd package update, and the default routes got set up normally on my test devices. Your fix works for me!

**Comment by feryw** [comment]
> This netifd update should fix it: https://git.openwrt.org/41d7439af5a40a71a0deb2075fa133b61b0e478a

After upgrade netifd i get this error
```
/etc/init.d/network restart
radio0(mac80211): Interface type not supported
```

**Comment by fgimenezm** [comment]
> This netifd update should fix it: https://git.openwrt.org/41d7439af5a40a71a0deb2075fa133b61b0e478a

Updated the netifd package to version 2023-11-07-516ab774-1 and now I have default gateway properly setup upon reboot (tested twice just in case) without any "option device" in /etc/config/network

So the update appears to have fixed the issue at least for me.

Thanks!

PS: I didn't try with custom interfase names though. I'll leave that test to someone else. I'm fine with the default names.

**Comment by fgimenezm** [comment]
Update: there seems to be some timing issue still there. Not sure if it's related though.

I have a guest wifi, currently on device "phy1-ap1", and the "guest" network doesn't start OK. It says "Error: Network device is not present".
This "guest" network is not bridged to anything, it's just for wireless guests.
If I restart the "guest" network (not the wireless, the network) then it starts OK.

relevant log entries (notice that the "phy1-ap1: link becomes ready" happens after the "guest" network has already failed.
```
Mon Nov 13 19:30:23 2023 daemon.notice netifd: Interface 'guest' is enabled
Mon Nov 13 19:30:23 2023 daemon.notice netifd: Interface 'guest' is setting up now
Mon Nov 13 19:30:23 2023 daemon.notice netifd: Interface 'guest' is now up
Mon Nov 13 19:30:23 2023 daemon.notice netifd: Interface 'guest' is now down
Mon Nov 13 19:30:23 2023 daemon.notice netifd: Interface 'guest' is disabled
Mon Nov 13 19:30:23 2023 kern.info kernel: [   80.069843] IPv6: ADDRCONF(NETDEV_CHANGE): phy1-ap1: link becomes ready
```

when I restart the "guest" network the log looks like this:
```
Mon Nov 13 20:15:07 2023 daemon.notice wpa_supplicant[1489]: Set new config for phy phy1
Mon Nov 13 20:15:07 2023 daemon.notice hostapd: Set new config for phy phy1: /var/run/hostapd-phy1.conf
Mon Nov 13 20:15:07 2023 daemon.notice hostapd: Reloaded settings for phy phy1
Mon Nov 13 20:15:07 2023 daemon.notice netifd: Wireless device 'radio1' is now up
Mon Nov 13 20:15:07 2023 daemon.notice netifd: Network device 'phy1-ap0' link is up
Mon Nov 13 20:15:07 2023 daemon.notice netifd: Network device 'phy1-ap1' link is up
Mon Nov 13 20:15:07 2023 daemon.notice netifd: Interface 'guest' is enabled
Mon Nov 13 20:15:07 2023 daemon.notice netifd: Interface 'guest' is setting up now
Mon Nov 13 20:15:07 2023 daemon.notice netifd: Interface 'guest' is now up
Mon Nov 13 20:15:07 2023 daemon.notice netifd: Interface 'guest' has link connectivity
Mon Nov 13 20:15:08 2023 user.notice firewall: Reloading firewall due to ifup of guest (phy1-ap1)
```

**Comment by fgimenezm** [comment]
Update 2: the issue with my guest wifi only happens when "wifi-device" has two "wifi-iface" (two SSIDs on the same radio device). Running "wifi reconf radio1" fixes it.
At this point I'm thinking this is not the same race condition bug that was causing the default gateway issue.

**Comment by brada4** [comment]
Issue? Just find a checkbox to up empty network.

**Comment by fgimenezm** [comment]
> Issue? Just find a checkbox to up empty network.

I've tried that. The "force link" on the network is there and I've also tried creating an empty bridge. With the empty bridge the network was up but the wireless didn't join the bridge. I had to run the "wifi reconf radio1" command to make it work.
Everything works OK if the wifi radio has only one SSID. The issue happens when there are two.
But as I said, it appears to be something in netifd but probably not the same as the original issue on this thread. I should probably open another issue.

**Comment by evoserver** [comment]
I tested version 23.05.2 and the problem is gone on Archer C60 v2. However, I keep on having problems with WDS which lets me attach one device on the same subnet with another acces point, so I'm still on RC3.
Thanks for your help and for solving this bug.

**Comment by fgimenezm** [comment]
My other problem with the Guest wifi is also resolved in 23.05.2.
Thanks everyone involved in making OpenWRT the best router firmware!

**Comment by brada4** [comment]
You use relayd, not wds, try to play through wds guide as your hardware is capable of it. @evoserver

**Comment by divergentti** [comment]
Is this problem solved? I upgraded to OpenWrt 23.05.5 r24106-10cc5fcd00 / LuCI openwrt-23.05 branch git-24.264.56413-c7a3562 and have now done at least 10 total resets and reconfigs, no success. Worst of all is that every time I test something, I loose connectivity and then reset everything and start over.

EA5300 is just switch and NAS. Ports 1-4 should be switched as before in 22 release. This seems to work OK. Devices in ports 1-4 are communicting each other just fine. However, for very strange reason, DNS resolver works just fine, but pings outside from local lan subnet do not. Pings to default gateway works ok.

This is seen as:

1. LuCi  update lists:   "Unable to execute opkg update command: SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data"

2. opkg update at CLI "Failed to send request: Operation not permitted" ** Failed to download the package list from https://downloads.openwrt.org/releases/23.05.5/packages/arm_cortex-a7_neon-vfpv4/base/Packages.gz

root@blaablaa:~# ip ro sh
default via 192.168.6.1 dev br-lan 
192.168.6.0/24 dev br-lan scope link  src 192.168.6.2 

ping 192.168.6.1
PING 192.168.6.1 (192.168.6.1): 56 data bytes
64 bytes from 192.168.6.1: seq=0 ttl=64 time=0.588 ms
64 bytes from 192.168.6.1: seq=1 ttl=64 time=0.426 ms
64 bytes from 192.168.6.1: seq=2 ttl=64 time=0.403 ms

ping downloads.openwrt.org
PING downloads.openwrt.org (151.101.2.132): 56 data bytes
22 packets transmitted, 0 packets received, 100% packet loss


cat /etc/config/network 

config interface 'loopback'
	option device 'lo'
	option proto 'static'
	option ipaddr '127.0.0.1'
	option netmask '255.0.0.0'

config globals 'globals'
	option ula_prefix 'fdd8:89f6:027b::/48'

config device
	option name 'br-lan'
	option type 'bridge'
	list ports 'eth0'
	list ports 'lan1'
	list ports 'lan2'
	list ports 'lan3'
	list ports 'lan4'
	list ports 'wan'
	option acceptlocal '1'
	option ipv6 '0'

config device
	option name 'lan1'
	option macaddr '58:ef:68:5e:0f:b9'

config device
	option name 'lan2'
	option macaddr '58:ef:68:5e:0f:b9'

config device
	option name 'lan3'
	option macaddr '58:ef:68:5e:0f:b9'

config device
	option name 'lan4'
	option macaddr '58:ef:68:5e:0f:b9'

config interface 'lan'
	option device 'br-lan'
	option proto 'static'
	option ipaddr '192.168.6.2'
	option netmask '255.255.255.0'
	option ip6assign '60'
	option gateway '192.168.6.1'
	list dns '192.168.6.3'
	option delegate '0'

config device
	option name 'wan'
	option macaddr '58:ef:68:5e:0f:b8'
	option ipv6 '0'

config interface 'wan'
	option proto 'dhcp'
	option auto '0'

config interface 'wan6'
	option proto 'dhcpv6'
	option auto '0'
	option reqaddress 'try'
	option reqprefix 'auto'


Perhaps I do not see some obvious setting which needs to be checked, but this could be also a bug. Thank you in advance for your opinion and advice.

**Comment by evoserver** [comment]
Hi Jari,

the problem with the default gateway was solved, but I had to remove my WDS configuration. However, I bought another router as my TP-Link Archer C60 was very old so I can't help you anymore on this issue.
Have a nice day,

Giovanni

On Nov 21, 2024 at 5:45 PM, Jari Hiltunen ***@***.***> wrote:

Is this problem solved? I upgraded to OpenWrt 23.05.5 r24106-10cc5fcd00 / LuCI openwrt-23.05 branch git-24.264.56413-c7a3562 and have now done at least 10 total resets and reconfigs, no success. Worst of all is that every time I test something, I loose connectivity and then reset everything and start over.
EA5300 is just switch and NAS. Ports 1-4 should be switched as before in 22 release. This seems to work OK. Devices in ports 1-4 are communicting each other just fine. However, for very strange reason, DNS resolver works just fine, but pings outside from local lan subnet do not. Pings to default gateway works ok.
This is seen as:
LuCi  update lists:   "Unable to execute opkg update command: SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data"
opkg update at CLI "Failed to send request: Operation not permitted" ** Failed to download the package list from https://downloads.openwrt.org/releases/23.05.5/packages/arm_cortex-a7_neon-vfpv4/base/Packages.gz
***@***.***:~# ip ro sh
default via 192.168.6.1 dev br-lan
192.168.6.0/24 dev br-lan scope link  src 192.168.6.2
ping 192.168.6.1
PING 192.168.6.1 (192.168.6.1): 56 data bytes
64 bytes from 192.168.6.1: seq=0 ttl=64 time=0.588 ms
64 bytes from 192.168.6.1: seq=1 ttl=64 time=0.426 ms
64 bytes from 192.168.6.1: seq=2 ttl=64 time=0.403 ms
ping downloads.openwrt.org
PING downloads.openwrt.org (151.101.2.132): 56 data bytes
22 packets transmitted, 0 packets received, 100% packet loss
cat /etc/config/network
config interface 'loopback'
option device 'lo'
option proto 'static'
option ipaddr '127.0.0.1'
option netmask '255.0.0.0'
config globals 'globals'
option ula_prefix 'fdd8:89f6:027b::/48'
config device
option name 'br-lan'
option type 'bridge'
list ports 'eth0'
list ports 'lan1'
list ports 'lan2'
list ports 'lan3'
list ports 'lan4'
list ports 'wan'
option acceptlocal '1'
option ipv6 '0'
config device
option name 'lan1'
option macaddr '58:ef:68:5e:0f:b9'
config device
option name 'lan2'
option macaddr '58:ef:68:5e:0f:b9'
config device
option name 'lan3'
option macaddr '58:ef:68:5e:0f:b9'
config device
option name 'lan4'
option macaddr '58:ef:68:5e:0f:b9'
config interface 'lan'
option device 'br-lan'
option proto 'static'
option ipaddr '192.168.6.2'
option netmask '255.255.255.0'
option ip6assign '60'
option gateway '192.168.6.1'
list dns '192.168.6.3'
option delegate '0'
config device
option name 'wan'
option macaddr '58:ef:68:5e:0f:b8'
option ipv6 '0'
config interface 'wan'
option proto 'dhcp'
option auto '0'
config interface 'wan6'
option proto 'dhcpv6'
option auto '0'
option reqaddress 'try'
option reqprefix 'auto'
Perhaps I do not see some obvious setting which needs to be checked, but this could be also a bug. Thank you in advance for your opinion and advice.

**Comment by divergentti** [comment]
Thank you Giovanni for your reply. Just after I hit "comment" I realized what really was wrong: I had another FW before the Internet access and IP of the OpenWRT box was of course filtered out :) ... so, all good, everything works just fine and thank you for your effort for the project!

73's

Jari
